asf_heat_pump_suitability/pipeline/prepare_features/garden_size.py

```diff
@@ -0,0 +1,125 @@
+import geopandas as gpd
+import logging
+import pandas as pd
+from collections.abc import Mapping
+
+
+def match_dict_files_land_building(
+    land_files_gdf: gpd.GeoDataFrame, building_files_gdf: gpd.GeoDataFrame
+) -> Mapping:
+    """
+    Get dictionary of intersecting INSPIRE land parcel files to Microsoft building footprint files.
+
+    Args:
+        land_files_gdf (gpd.GeoDataFrame): INSPIRE land file names and file bounding polygons
+        building_files_gdf (gpd.GeoDataFrame): Microsoft building footprint file names and file bounding polygons
+
+    Returns:
+        Mapping: mapping where keys are INSPIRE land parcel file names and values are Microsoft building footprint file
+        names
+    """
+    logging.info("Mapping building footprint files to INSPIRE land registry files")
+    gdf = gpd.overlay(land_files_gdf, building_files_gdf, how="intersection")
```

Comment:
This should be:
```python
gdf = gpd.sjoin(land_files_gdf, building_files_gdf, predicate="intersects")
```
It produces the same result as overlay much more cheaply.

Basically, spatial joins (sjoins) join tables on the basis of a spatial relationship, but they don't change any of the actual geometries. The overlay operation does this, plus it create a new geometry based on the spatial operation.

So, sjoin will return matching records for an intersection and the _original_ geometries. Whereas, overlay will return matching records for an intersection and a new geometry that describe that intersection only - like with sets.

In this case you only want to understand the spatial relationships, not the actual intersection, so overlay is overkill.